File: GestionTicketsAPI/DTOs/ClientUpdateDto.cs
using System;

namespace GestionTicketsAPI.DTOs;

public class ClientUpdateDto
{
  public int Id { get; set; }
  public required string Email { get; set; }
  public required string FirstName { get; set; }
  public required string LastName { get; set; }
  public required string NumTelephone { get; set; }
  public required int PaysId { get; set; } 
  public required bool Actif { get; set; }
  public int SocieteId { get; set; }

  public string? NouveauPassword { get; set; }
  public string? ConfirmNouveauPassword { get; set; }
}

File: gestion-tickets/src/app/DTOs/ClientUpdateDto.ts
export interface ClientUpdateDto {
  id:            number;
  email:         string;
  firstName:     string;
  lastName:      string;
  numTelephone:  string;
  paysId:        number;   // le code du pays (ex. `"1"`)
  actif:         boolean;
  societeId:     number;
  nouveauPassword?:          string;
  confirmNouveauPassword?:   string;
}
